src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1941,7 +1941,7 @@ pub trait DecimalDecoder: BytesDecoder {
             if d.word_buf[word_idx] >= TEN_POW[leading_digits + 1] {
                 return Err(box_err!("invalid leading digits for decimal number"));
             }
-            if d.word_buf[word_idx] != 0 {
+            if word_idx > 0 || d.word_buf[word_idx] != 0 {
```

Comment:
When will `word_idx > 0`?

Reply:
This expression is always false, but it is more independent, it doesn't depend on the fact that word_idx is initialized to 0.

Reply:
What can be benefit from this? Can the function still work if `word_idx` is initialized to non-zero?